src/qibocal/auto/operation.py

```diff
@@ -98,68 +97,96 @@ def load(cls, input_parameters):
         return instantiated_class
 
 
-class Data:
-    """Data resulting from acquisition routine."""
+class AbstractData:
+    """Abstract data class."""
 
-    data: dict[Union[tuple[QubitId, int], QubitId], npt.NDArray]
-    """Data object to store arrays"""
-
-    @property
-    def qubits(self):
-        """Access qubits from data structure."""
-        if set(map(type, self.data)) == {tuple}:
-            return list({q[0] for q in self.data})
-        return [q for q in self.data]
-
-    @property
-    def pairs(self):
-        """Access qubit pairs ordered alphanumerically from data structure."""
-        return list({tuple(sorted(q[:2])) for q in self.data})
+    def __init__(
+        self, data: dict[Union[tuple[QubitId, int], QubitId], npt.NDArray] = None
+    ):
+        self.data = data if data is not None else {}
 
     def __getitem__(self, qubit: Union[QubitId, tuple[QubitId, int]]):
         """Access data attribute member."""
         return self.data[qubit]
 
     @property
-    def global_params(self) -> dict:
+    def params(self) -> dict:
         """Convert non-arrays attributes into dict."""
         global_dict = asdict(self)
-        global_dict.pop("data")
+        if hasattr(self, "data"):
+            global_dict.pop("data")
         return global_dict
 
-    def save(self, path):
-        """Store results."""
-        self._to_json(path)
-        self._to_npz(path)
+    def save(self, path: Path, filename: str):
+        """Dump class to file."""
+        self._to_json(path, filename)
+        self._to_npz(path, filename)
 
-    def _to_npz(self, path):
+    def _to_npz(self, path: Path, filename: str):
         """Helper function to use np.savez while converting keys into strings."""
-        np.savez(path / DATAFILE, **{json.dumps(i): self.data[i] for i in self.data})
+        if hasattr(self, "data"):
+            np.savez(
+                path / filename,
```

Comment:
```suggestion
                path / f"{filename}.npz",
```
I think it is better to explicitly sate the extension, even if numpy adds it automatically.